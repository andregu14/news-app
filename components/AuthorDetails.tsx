import { View, ViewProps, Text } from "./Themed";
import { Image } from "expo-image";
import { StyleSheet, useColorScheme } from "react-native";
import Colors from "@/constants/Colors";

type AuthorDetailsProps = ViewProps & {
  name: string;
  photo?: string;
  date: string;
  department: string;
  imageTestId?: string
};

export default function AuthorDetails({
  name,
  photo,
  date,
  department,
  imageTestId,
  ...otherProps
}: AuthorDetailsProps) {
  const colorScheme = useColorScheme() ?? "light";
  const themeColors = Colors[colorScheme];

  return (
    <View style={[styles.container, otherProps.style]}>
      <View
        style={[
          styles.photoContainer,
          {
            borderColor:
              colorScheme === "dark" ? "rgba(255, 255, 255, 0.15)" : "#eee",
          },
        ]}
      >
        <Image
          source={{ uri: photo }}
          placeholder={require("@/assets/images/photo-placeholder.png")}
          style={styles.photo}
          contentFit="cover"
          testID={imageTestId}
        />
      </View>
      <View>
        <Text style={[styles.text, { fontWeight: "bold" }]}>{name}</Text>
        <Text
          style={[styles.subText, { color: themeColors.secondaryText }]}
        >{`${date[0].toUpperCase() + date.slice(1)} â€¢ Em ${department}`}</Text>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    alignItems: "center",
    marginBottom: 10,
  },
  photoContainer: {
    width: 40,
    height: 40,
    borderRadius: 20,
    marginRight: 10,
    borderWidth: 2,
    overflow: "hidden",
  },
  photo: {
    width: "100%",
    height: "100%",
    backgroundColor: "#bcbdc1",
  },
  text: {
    fontSize: 15,
  },
  subText: {
    fontSize: 13,
  },
});
