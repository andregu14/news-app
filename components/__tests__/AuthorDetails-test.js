import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react-native';
import AuthorDetails from '../AuthorDetails';

jest.mock("react-native/Libraries/Utilities/useColorScheme", () => {
    return {
        default: jest.fn(() => "light")
    }
})

describe('<AuthorDetails />', () => {
    it('renders correctly with name, date, department props and matches snapshot', () => {
        const testName = "Test Name"
        const testDate = "há 7 dias"
        const testDepartment = "Test Department"
        render(<AuthorDetails name={testName} date={testDate} department={testDepartment} />);
        expect(screen.getByText(testName)).toBeVisible();
        expect(screen.getByText(`${testDate[0].toUpperCase() + testDate.slice(1)} • Em ${testDepartment}`)).toBeVisible();
        expect(screen.toJSON()).toMatchSnapshot();
    });

    it('renders correctly with photo prop and matches snapshot', () => {
        const testImage = "https://example.com/test-image.png"
        render(<AuthorDetails photo={testImage} imageTestId='author-image' name='test' date='test' department='test' />)
        const imageComponent = screen.getByTestId('author-image');
        expect(imageComponent).toBeVisible();
        expect(imageComponent.props.source[0].uri).toEqual(testImage);
        expect(screen.toJSON()).toMatchSnapshot();
    })

    it('renders correctly with placeholder photo and matches snapshot', () => {
        render(<AuthorDetails name='name' date='test' department='test' imageTestId='default-author-image' />)
        const imageComponent = screen.getByTestId('default-author-image')
        expect(imageComponent).toBeVisible();
        expect(imageComponent.props.placeholder[0].testUri).toEqual("../../../assets/images/photo-placeholder.png");
        expect(screen.toJSON()).toMatchSnapshot();
    })

})