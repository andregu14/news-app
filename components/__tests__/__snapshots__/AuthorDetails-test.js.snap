// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthorDetails /> renders correctly with name, date, department props and matches snapshot 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      [
        {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 10,
        },
        undefined,
      ],
    ]
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "#fff",
        },
        [
          {
            "borderRadius": 20,
            "borderWidth": 2,
            "height": 40,
            "marginRight": 10,
            "overflow": "hidden",
            "width": 40,
          },
          {
            "borderColor": "#eee",
          },
        ],
      ]
    }
  >
    <ViewManagerAdapter_ExpoImage
      backgroundColor={4290559425}
      containerViewRef={"[React.ref]"}
      contentFit="cover"
      contentPosition={
        {
          "left": "50%",
          "top": "50%",
        }
      }
      height="100%"
      nativeViewRef={"[React.ref]"}
      onError={[Function]}
      onLoad={[Function]}
      onLoadStart={[Function]}
      onProgress={[Function]}
      placeholder={
        [
          {
            "testUri": "../../../assets/images/photo-placeholder.png",
          },
        ]
      }
      source={
        [
          {
            "uri": undefined,
          },
        ]
      }
      style={
        {
          "backgroundColor": "#bcbdc1",
          "height": "100%",
          "width": "100%",
        }
      }
      transition={null}
      width="100%"
    />
  </View>
  <View
    style={
      [
        {
          "backgroundColor": "#fff",
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#000",
          },
          [
            {
              "fontSize": 15,
            },
            {
              "fontWeight": "bold",
            },
          ],
        ]
      }
    >
      Test Name
    </Text>
    <Text
      style={
        [
          {
            "color": "#000",
          },
          [
            {
              "fontSize": 13,
            },
            {
              "color": "#888",
            },
          ],
        ]
      }
    >
      Há 7 dias • Em Test Department
    </Text>
  </View>
</View>
`;

exports[`<AuthorDetails /> renders correctly with photo prop and matches snapshot 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      [
        {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 10,
        },
        undefined,
      ],
    ]
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "#fff",
        },
        [
          {
            "borderRadius": 20,
            "borderWidth": 2,
            "height": 40,
            "marginRight": 10,
            "overflow": "hidden",
            "width": 40,
          },
          {
            "borderColor": "#eee",
          },
        ],
      ]
    }
  >
    <ViewManagerAdapter_ExpoImage
      backgroundColor={4290559425}
      containerViewRef={"[React.ref]"}
      contentFit="cover"
      contentPosition={
        {
          "left": "50%",
          "top": "50%",
        }
      }
      height="100%"
      nativeViewRef={"[React.ref]"}
      onError={[Function]}
      onLoad={[Function]}
      onLoadStart={[Function]}
      onProgress={[Function]}
      placeholder={
        [
          {
            "testUri": "../../../assets/images/photo-placeholder.png",
          },
        ]
      }
      source={
        [
          {
            "uri": "https://example.com/test-image.png",
          },
        ]
      }
      style={
        {
          "backgroundColor": "#bcbdc1",
          "height": "100%",
          "width": "100%",
        }
      }
      testID="author-image"
      transition={null}
      width="100%"
    />
  </View>
  <View
    style={
      [
        {
          "backgroundColor": "#fff",
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#000",
          },
          [
            {
              "fontSize": 15,
            },
            {
              "fontWeight": "bold",
            },
          ],
        ]
      }
    >
      test
    </Text>
    <Text
      style={
        [
          {
            "color": "#000",
          },
          [
            {
              "fontSize": 13,
            },
            {
              "color": "#888",
            },
          ],
        ]
      }
    >
      Test • Em test
    </Text>
  </View>
</View>
`;

exports[`<AuthorDetails /> renders correctly with placeholder photo and matches snapshot 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      [
        {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 10,
        },
        undefined,
      ],
    ]
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "#fff",
        },
        [
          {
            "borderRadius": 20,
            "borderWidth": 2,
            "height": 40,
            "marginRight": 10,
            "overflow": "hidden",
            "width": 40,
          },
          {
            "borderColor": "#eee",
          },
        ],
      ]
    }
  >
    <ViewManagerAdapter_ExpoImage
      backgroundColor={4290559425}
      containerViewRef={"[React.ref]"}
      contentFit="cover"
      contentPosition={
        {
          "left": "50%",
          "top": "50%",
        }
      }
      height="100%"
      nativeViewRef={"[React.ref]"}
      onError={[Function]}
      onLoad={[Function]}
      onLoadStart={[Function]}
      onProgress={[Function]}
      placeholder={
        [
          {
            "testUri": "../../../assets/images/photo-placeholder.png",
          },
        ]
      }
      source={
        [
          {
            "uri": undefined,
          },
        ]
      }
      style={
        {
          "backgroundColor": "#bcbdc1",
          "height": "100%",
          "width": "100%",
        }
      }
      testID="default-author-image"
      transition={null}
      width="100%"
    />
  </View>
  <View
    style={
      [
        {
          "backgroundColor": "#fff",
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#000",
          },
          [
            {
              "fontSize": 15,
            },
            {
              "fontWeight": "bold",
            },
          ],
        ]
      }
    >
      name
    </Text>
    <Text
      style={
        [
          {
            "color": "#000",
          },
          [
            {
              "fontSize": 13,
            },
            {
              "color": "#888",
            },
          ],
        ]
      }
    >
      Test • Em test
    </Text>
  </View>
</View>
`;
